File: TYVObjCHomework/Source/TYVDispatch/TYVDispatch.c
//
//  TYVDispatch.c
//  IDAPCoursesHomework
//
//  Created by Yurii Trokhymchuk on 6/9/15.
//  Copyright (c) 2015 YURII. All rights reserved.
//

#include "TYVDispatch.h"

#pragma mark -
#pragma mark Private Declarations

typedef void(TYVperfomBlockOnMainQueue)(void *self, void *context);

#pragma mark -
#pragma mark Public Implementations

void dispatch_sync_onMainQueue(TYVperfomBlockOnMainQueue *block) {

}

File: TYVObjCHomework/Source/TYVDispatch/TYVDispatch.m
//
//  TYVDispatch.m
//  IDAPCoursesHomework
//
//  Created by YURII on 10.06.15.
//  Copyright (c) 2015 YURII. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "TYVDispatch.h"
#import "TYVProtocolObservableObject.h"

#pragma mark -
#pragma mark Public Implementations

void TYVDispatchSyncOnMainQueueWithBlock(TYVBlock block) {
    if ([NSThread isMainThread]) {
        block();
    } else {
        dispatch_sync(dispatch_get_main_queue(), block);
    }
}
